### PR TITLE
Quartz scheduling

### PR DESCRIPTION
**Pull request by YevheniiChornyi**

### CODE REVIEW

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class LostfilmRssJob implements Job {
+
+
+    private final FeedService feedService;
+    private final TVSeriesConvertor convertService;
+    private final TVSeriesService seriesService;
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        String url = "https://www.lostfilm.tv/rss.xml";
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: move this url value to properties file

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        for (FeedMessage fs :
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: use streams

File: src/main/java/yevhenii/lostfilmdemo/services/impl/SchedulerServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+@Service
+public class SchedulerServiceImpl implements SchedulerService {
+    @Getter
+    static int jobCount;
+    @Getter
+    static int triggerCount;
+
+    public void start() throws SchedulerException {
+        Scheduler scheduler = StdSchedulerFactory.getDefaultScheduler();
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/services/impl/SchedulerServiceImpl.java]: you don't need to do that, you are using starter and it has auto configuration

Comment (YevheniiChornyi) [reply src/main/java/yevhenii/lostfilmdemo/services/impl/SchedulerServiceImpl.java]: oh..tu, cause i wasting few hours to find how to use di here

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        log.info("on execute");
+        feedService.readFeed(getUrl())
+                .forEach(fs -> seriesService.save(convertService.convert(fs)));
+        log.info("end executing");
+    }
+
+    private String getUrl(){
+        String propertyFilePath = "config.properties";
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: place your property inside application.properties and here just inject it using @Value annotation

File: src/main/java/yevhenii/lostfilmdemo/services/impl/SchedulerServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    static int jobCount;
+    @Getter
+    static int triggerCount;
+    private final Scheduler scheduler;
+
+    public void start() throws SchedulerException {
+
+        Trigger trigger = getTrigger(getJobDetail());
+        if (!scheduler.checkExists(trigger.getKey()))
+        scheduler.scheduleJob(trigger);
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/services/impl/SchedulerServiceImpl.java]: you don't need this as well.

https://github.com/kenshoo/fb-dca-template-manager/blob/master/application/src/main/java/io/skai/fb/dca/template/quartz/config/JobDetailsCfg.java

https://github.com/kenshoo/fb-dca-template-manager/blob/master/application/src/main/java/io/skai/fb/dca/template/quartz/job/ExampleJob.java

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1,4 +1,9 @@
...
 spring.datasource.username=root
 spring.datasource.password=root2021
-spring.liquibase.change-log=db/changelog/changelog_beta.xml
\ No newline at end of file
+spring.liquibase.change-log=db/changelog/changelog_beta.xml
+spring.sql.init.mode=never
+spring.quartz.job-store-type=jdbc
+spring.quartz.jdbc.initialize-schema=never
+
+tvSeries.url = https://www.lostfilm.tv/rss.xml
```
Comment (mmarusich) [inline src/main/resources/application.properties]: rename to `lostfilm.rss.url`

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import org.quartz.JobExecutionException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import yevhenii.lostfilmdemo.convertors.TVSeriesConvertor;
+import yevhenii.lostfilmdemo.services.FeedService;
+import yevhenii.lostfilmdemo.services.TVSeriesService;
+
+@Log4j2
+@Component
+@RequiredArgsConstructor//(onConstructor = @__(@Autowired))
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: autowired redundant

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import yevhenii.lostfilmdemo.convertors.TVSeriesConvertor;
+import yevhenii.lostfilmdemo.services.FeedService;
+import yevhenii.lostfilmdemo.services.TVSeriesService;
+
+@Log4j2
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: use Sl4j annotation instead

File: src/main/java/yevhenii/lostfilmdemo/quartz/JobConfig.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    private static int jobCount = 1;
+    @Getter
+    private static int triggerCount = 1;
+
+    @Bean
+    public JobDetail getJobDetail() {
+        log.info("creating job");
+        return JobBuilder.newJob(LostfilmRssJob.class)
+                .storeDurably()
+                .withIdentity("lostfilm_job" + jobCount++, "lostfilmGroup")
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/JobConfig.java]: check this example how properly define identity https://github.com/kenshoo/fb-product-fetcher/blob/master/application/src/main/java/io/skai/fbproductfetcher/quartz/config/QuartzJobConfig.java#L19

No need adding any counters you did here

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+@Component
+@RequiredArgsConstructor//(onConstructor = @__(@Autowired))
+public class LostfilmRssJob implements Job {
+
+    private final FeedService feedService;
+    private final TVSeriesConvertor convertService;
+    private final TVSeriesService seriesService;
+    @Value("${tvSeries.url:https://www.lostfilm.tv/rss.xml}")
+    String url;
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: add private final, keep same code style

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+@Log4j2
+@Component
+@RequiredArgsConstructor//(onConstructor = @__(@Autowired))
+public class LostfilmRssJob implements Job {
+
+    private final FeedService feedService;
+    private final TVSeriesConvertor convertService;
+    private final TVSeriesService seriesService;
+    @Value("${tvSeries.url:https://www.lostfilm.tv/rss.xml}")
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: remove this default value

File: src/main/java/yevhenii/lostfilmdemo/services/impl/SchedulerServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class SchedulerServiceImpl implements SchedulerService {
+
+    private final Scheduler scheduler;
+
+    public void start() throws SchedulerException {
+
+        scheduler.start();
```
Comment (mmarusich) [inline src/main/java/yevhenii/lostfilmdemo/services/impl/SchedulerServiceImpl.java]: this also redundant, you can define in properties to make quartz start automatically

File: src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java (hunk 10 above)
Comment (YevheniiChornyi) [reply src/main/java/yevhenii/lostfilmdemo/quartz/LostfilmRssJob.java]: then it conflict with @RequiredArgsConstructor

**Review by mmarusich — APPROVED** [review]

**Pull request merged**